src/app.ts

```diff
@@ -212,19 +226,48 @@ export class App<State> {
     // - `app.mountApp("/*", otherApp)`
     const isSelf = path === "/*" || path === "/";
     if (isSelf && middlewares.length > 0) {
-      this.#router._middlewares.push(...middlewares);
-    }
+      // When mounting at the root, add middleware directly to the host app
+      for (const middleware of middlewares) {
+        this.#router._middlewares.push(middleware);
+      }
 
-    for (let i = 0; i < routes.length; i++) {
-      const route = routes[i];
+      // Add routes as-is since we're mounting at the root
+      for (let i = 0; i < routes.length; i++) {
+        const route = routes[i];
```

Comment:
```suggestion
      for (const route of routes) {
```
Nit. And ditto for other loops.

Reply:
It's fine as is. Which for-loop to use doesn't matter. Although it's unlikely to matter here, the traditional for-loop is even faster as it doesn't need to invoke the iterator stuff compared to `for-of`.